### ISSUE TITLE
Deprecate `ContainedList.ContainedListItem` usage

### ISSUE THREAD

**Issue by tay1orjones** [carbon-design-system/carbon#16354]
Surfaced in https://github.com/carbon-design-system/carbon/pull/16302#discussion_r1594042362, `ContainedListItem` is a field on `ContainedList`. The convention so far in v11 is to avoid adding subcomponents as fields on parent/sibling component objects/exports.

```[tasklist]
- [x] Use `deprecateFieldOnObject` to deprecate `ContainedList.ContainedListItem` | covered in #16552
- [x] Document this change in ContainedList.mdx storybook docs | covered in #16640
- [ ] Document the convention of avoiding subcomponents as fields in the developer style guide documentation in this repo
```

**Comment by Gururajj77** [comment]
Completed via #16552

**Comment by Gururajj77** [comment]
@tay1orjones , where should I document about the last part? not able to find developer style guild.

**Comment by Gururajj77** [comment]
closed via #16640
